### PR TITLE
commit final solution for 2765

### PR DESCRIPTION
**Pull request by OwenTian**

### CODE REVIEW

File: 2765. Longest Alternating Subarray/solution.py [hunk 1]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
 
     	# declare a subArraySize variable that keeps track of length of
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:11]: In your comments, you should explain what the variable keeps track of, which you did here, but you don't need to mention the name.

File: 2765. Longest Alternating Subarray/solution.py [hunk 2]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
 
     	# declare a subArraySize variable that keeps track of length of 
     	# alternating subarrays.
+        subArraySize = 0
 
-    	# declare a list to hold potential subArraySizes.
-
+    	# variable to hold the largest size
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:15]: In our language, we often don't use "variable" for comments. A comment isn't actually necessary here. Maybe "keeps track of the largest subarray size" is appropriate, but "largest size" is implied by the variable name.

File: 2765. Longest Alternating Subarray/solution.py [hunk 3]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+        largestSize = 0
+		# variable to keep track of index in found subarray
+        subArrayIndex = 0
+        
+        # boolean to determine if found sub array in while loop iteration
+        foundSubArray = False
+        
     	# loop through the array starting from 0, increment index by 1 
     	# each time. this loop is the outer loop.
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:24]: If you look at my commit, you'll see how I used just 1 loop by taking advantage of the fact that 2 subarrays can only share 1 element. For example, [2, 3, 4, 3] shows how only one element can overlap between two sequences [2, 3] and [3, 4, 3] which in this case is 3.

File: 2765. Longest Alternating Subarray/solution.py [hunk 4]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+                    # less than previous
+                    if subArraySize % 2 != 0 and \
+                    nums[subArrayIndex] == nums[subArrayIndex - 1] + 1:
+                        # increment subArraySize and subArrayIndex and continue
+                        subArraySize = subArraySize + 1
+                        subArrayIndex = subArrayIndex + 1
+                        foundSubArray = True
+                        continue
+                    
+                    # if we're at an odd counter, check if current int is one
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:51]: nit: even not odd

File: 2765. Longest Alternating Subarray/solution.py [hunk 5]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+        
+        # boolean to determine if found sub array in while loop iteration
+        foundSubArray = False
+        
     	# loop through the array starting from 0, increment index by 1 
     	# each time. this loop is the outer loop.
+        i = 0
+        while i < len(nums) - 1:
 
     		# if the int at next index is one greater than int at the
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:28]: nit: extra indent

File: 2765. Longest Alternating Subarray/solution.py [hunk 6]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
 
     	# declare a subArraySize variable that keeps track of length of 
     	# alternating subarrays.
+        subArraySize = 0
 
-    	# declare a list to hold potential subArraySizes.
-
+    	# variable to hold the largest size
+        largestSize = 0
+		# variable to keep track of index in found subarray
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:17]: nit: extra indent

File: 2765. Longest Alternating Subarray/solution.py [hunk 7]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+            if nums[i] + 1 == nums[i+1]:
+                
+                # set the sub array size to 2 since we have 2 elements
+                subArraySize = 2
+                
+                # set subArrayIndex to keep track of where we are in sub array,
+                # add 2 since we're moving forward past the two elements we know
+                subArrayIndex = i + 2
+      
     			# check for continuation of the subarray via:
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:39]: nit: extra indent

File: 2765. Longest Alternating Subarray/solution.py [hunk 8]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+                        break
 
-    			# while current value (at subArrayIndex) - 1 = next value and 
-    			# current value = next next value and subArrayIndex 
-    			# (the current index) is less than length of nums - 2:
-
-    				# increment subArraySize by 2
-    				# increment subArrayIndex by 2
+                                        
+    			# reached end of sub array. record subArraySize if it's larger
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:67]: nit: extra indent

File: 2765. Longest Alternating Subarray/solution.py [hunk 9]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
 
     	# declare a subArraySize variable that keeps track of length of 
     	# alternating subarrays.
+        subArraySize = 0
```
Comment (winry-fruitiwi) [inline 2765. Longest Alternating Subarray/solution.py:13]: nit: might be better as "alternatingSubLen"

File: 2765. Longest Alternating Subarray/solution.py [hunk 10]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+                
+                # set subArrayIndex to keep track of where we are in sub array,
+                # add 2 since we're moving forward past the two elements we know
+                subArrayIndex = i + 2
+      
     			# check for continuation of the subarray via:
+                while subArrayIndex < len(nums):
+                    # if we're at an odd counter, check if current int is one 
+                    # less than previous
+                    if subArraySize % 2 != 0 and \
```
Comment (cody-berry) [inline 2765. Longest Alternating Subarray/solution.py:43]: I didn't realize you could use backslash to join 2 lines! I just used parens.

**Review by cody-berry — CHANGES_REQUESTED** [review]
Mostly just extra indents and commenting variables. I don't need you to, but I'd recommend you change these comments to help you understand comments. Otherwise, I like your backslash and excellent!

File: 2765. Longest Alternating Subarray/solution.py (hunk 4 above)
Comment (winry-fruitiwi) [reply 2765. Longest Alternating Subarray/solution.py:51]: Actually you can just have a "diff" variable equal to 1 or -1, then multiply it by -1 each time you continue an alternating subarray.

File: 2765. Longest Alternating Subarray/solution.py [hunk 11]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+                    nums[subArrayIndex] == nums[subArrayIndex - 1] + 1:
```
Comment (aeruru) [inline 2765. Longest Alternating Subarray/solution.py:44]: By saving nums[subArrayIndex] - nums[subArrayIndex - 1] into a variable, you could have combined your two if clauses here and gotten rid of the code duplication where you increment the subArraySize

File: 2765. Longest Alternating Subarray/solution.py [hunk 12]
```diff
@@ -11,39 +10,81 @@ def alternatingSubarray(self, nums: List[int]) -> int:
...
+                    if subArraySize % 2 == 0 and \
+                    nums[subArrayIndex] == nums[subArrayIndex - 1] - 1:
+                        # increment subArraySize and subArrayIndex and continue
+                        subArraySize = subArraySize + 1
+                        subArrayIndex = subArrayIndex + 1
+                        foundSubArray = True
+                        continue
+                        
+                    # else break out of while loop
+                    else:
```
Comment (aeruru) [inline 2765. Longest Alternating Subarray/solution.py:62]: This else is extraneous because of the continue in the if clause.

**Review by aeruru — APPROVED** [review]

**Pull request merged**